internal/services/nginx/nginx_deployment_data_source.go

```diff
@@ -194,10 +196,74 @@ func (m DeploymentDataSource) Attributes() map[string]*pluginsdk.Schema {
 			Computed: true,
 		},
 
+		"web_application_firewall_settings": {
```

Comment:
@katbyte I have been following the approach of keeping the provider consistent with the swagger spec. Which is the reason for using `web_application_firewall_settings`, `activation_state` and having the `activation_state` be an enum instead of a bool as per the [swagger](https://github.com/Azure/azure-rest-api-specs/blob/fa6c6015b1633a701edb629c1a759433aa1fdaf2/specification/nginx/resource-manager/NGINX.NGINXPLUS/preview/2024-09-01-preview/swagger.json#L2040) spec. I think deviating from the swagger spec will causes more confusion for users so would prefer to use the same naming. Do let me know if provider has a different set of naming convention that it needs to follow.

Reply:
In our contribution guide we have a section on naming https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-naming.md

the convention is to generally remove things like "settings" "configuration"